scripts/obico.sh

```diff
@@ -0,0 +1,433 @@
+      done && select_msg "${new_moonraker_obico_count}"
+    else
+      log_error "Internal error. moonraker_count of '${moonraker_count}' not equal or grather than one!"
+      return 1
+    fi  # (( moonraker_count == 1 ))
+
+    ### Step 2: Confirm instance amount
+    local yn
+    while true; do
+      (( new_moonraker_obico_count == 1 )) && local question="Install Moonraker-obico?"
+      (( new_moonraker_obico_count > 1 )) && local question="Install ${new_moonraker_obico_count} Moonraker-obico instances?"
+      read -p "${cyan}###### ${question} (Y/n):${white} " yn
+      case "${yn}" in
+        Y|y|Yes|yes|"")
+          select_msg "Yes"
+          break;;
+        N|n|No|no)
+          select_msg "No"
+          abort_msg "Exiting Moonraker-obico setup ...\n"
+          return;;
+        *)
+          error_msg "Invalid Input!";;
+      esac
+    done
+  fi  # (( allowed_moonraker_obico_count > 0 ))
+
+  if (( new_moonraker_obico_count > 0 )); then
+
+    ### Step 3: Ask for the Obico server URL
+    obico_server_url_prompt
+    local obico_server_url
+    while true; do
+      read -p "${cyan}###### Obico Server URL:${white} " -i "https://app.obico.io" -e obico_server_url
+      if echo "${obico_server_url}" | grep -qE "^(http|https)://[a-zA-Z0-9./?=_%:-]*"; then
+        break
+      else
+        error_msg "Invalid server URL!"
+      fi
+    done
+
+    (( new_moonraker_obico_count > 1 )) && status_msg "Installing ${new_moonraker_obico_count} Moonraker-obico instances ..."
+    (( new_moonraker_obico_count == 1 )) && status_msg "Installing Moonraker-obico ..."
+
+    ### Step 4: Install dependencies
+    local dep=(git dfu-util virtualenv python3 python3-pip python3-venv ffmpeg)
+    dependency_check "${dep[@]}"
+
+    ### Step 5: Clone the moonraker-obico repo
+    clone_moonraker_obico "${MOONRAKER_OBICO_REPO}"
+
+    ### step 6: call moonrake-obico/install.sh with the correct params
+    local port=7125 moonraker_cfg_dirs=($(get_config_folders))
+
+    if (( moonraker_count == 1 )); then
+      "${MOONRAKER_OBICO_DIR}/install.sh" -C "${moonraker_cfg_dirs[0]}/moonraker.conf" -p "${port}" -H 127.0.0.1 -l "${KLIPPER_LOGS}" -s -L -S "${obico_server_url}"
+    elif (( moonraker_count > 1 )); then
+      local j=${existing_moonraker_obico_count}
+
+      for (( i=1; i <= new_moonraker_obico_count; i++ )); do
+        "${MOONRAKER_OBICO_DIR}/install.sh" -n "${moonraker_names[${j}]}" -C "${moonraker_cfg_dirs[${j}]}/moonraker.conf" -p $((port+j)) -H 127.0.0.1 -l "${KLIPPER_LOGS}" -s -L -S "${obico_server_url}"
+        j=$(( j + 1 ))
+      done && unset j
+    fi # (( moonraker_count == 1 ))
+  fi  # (( new_moonraker_obico_count > 0 ))
+
+  ### Step 7: Link to the Obico server if necessary
+  local instance_name
+  local not_linked_instances=()
+  if (( moonraker_count == 1 )); then
+    if moonraker_obico_needs_linking "$(moonraker_obico_config 0)"; then
```

Comment:
The `moonraker_obico_config` function defined in line 24 does not take in any arguments. So the `0` you pass there (and a few lines below the `${i}`) won't get used inside the function. Im not quite sure what the intention was, if the arguments are obsolete or if the `moonraker_obico_config` function is incomplete?

Reply:
This is a really good catch! It it supposed to use `${1}`, not `${i}`. Not sure how my testing passed... Fixing it now.

Reply:
Should be fixed in [276c123](https://github.com/th33xitus/kiauh/pull/227/commits/276c123d3d6d0d3609e0f4b179b545d7d53f40cd). `moonraker_obico_config` should take an argument, which is the index of a specific instance in `multi_instance_names`